### PR TITLE
367. SwiftUI "Contests" screen.

### PR DESCRIPTION
**Pull request by derevyanko**

### CODE REVIEW

File: ios/Codeforces Watcher/Application/core/extensions/String+Extension.swift [hunk 1]
```diff
@@ -27,13 +27,23 @@ extension String {
...
         return self.attributed.apply {
             $0.addAttribute(NSAttributedString.Key.foregroundColor, value: color, range: NSRange(location: 0, length: self.count))
         }
     }
+    
+    func capitalizingFirstLetter() -> String {
+        prefix(1).capitalized + dropFirst()
+    }
+
+    mutating func capitalizeFirstLetter() {
```
Comment (yev-kanivets) [inline ios/Codeforces Watcher/Application/core/extensions/String+Extension.swift:44]: Wow, nice idea 👍

File: ios/Codeforces Watcher/Application/features/contests/ContestsView.swift [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+    
+    var onContest: (Contest) -> Void = { _ in }
+    var onCalendar: (Contest) -> Void = { _ in }
+    var onFilter: () -> Void = {}
+    var refreshControl = UIRefreshControl()
+    
+    private let noContestsExplanation = "Contests are on the way to your device..."
+    
+    var body: some View {
+        ZStack {
```
Comment (yev-kanivets) [inline ios/Codeforces Watcher/Application/features/contests/ContestsView.swift:16]: Why do we need a wrapping `ZStack`?

File: ios/Codeforces Watcher/Application/features/contests/ContestsView.swift [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+            }, label: {
+                Image("calendarAddIcon")
+                    .frame(width: 18, height: 20)
+            })
+        }
+        .padding(.horizontal, 20)
+        .padding(.vertical, 10)
+    }
+    
+    struct UIModel {
```
Comment (yev-kanivets) [inline ios/Codeforces Watcher/Application/features/contests/ContestsView.swift:97]: In `UiModel` you should have only the data you use in the view, so passing `Contest` is a bit overkill.

**Review by yev-kanivets — COMMENTED** [review]
Wow, very nice icons ✨

File: ios/Codeforces Watcher/Application/features/contests/ContestsView.swift (hunk 2 above)
Comment (derevyanko) [reply ios/Codeforces Watcher/Application/features/contests/ContestsView.swift:16]: Hm, really not unnecessary.

File: ios/Codeforces Watcher/Application/features/shared/NavigationBar.swift [hunk 4]
```diff
@@ -0,0 +1,36 @@
+import SwiftUI
+
+struct NavigationBar: View {
```
[the remark this answers is not in the document]
Comment (derevyanko) [reply ios/Codeforces Watcher/Application/features/shared/NavigationBar.swift:3]: I thought the same too, but the ContestsNavigationBar(in the next task) has a different concept, so in this case it takes a lot of parameters to be sent to this view and will be so massive.
Unfortunately, in this case it is better to have an own navigation bar for each screen.

File: ios/Codeforces Watcher/Application/features/contests/ContestsView.swift (hunk 3 above)
Comment (derevyanko) [reply ios/Codeforces Watcher/Application/features/contests/ContestsView.swift:97]: Contests is passed for `onCalendar` and `onContest` functions, which use the required fields.
Also, I can create a struct with required fields for the view and this functions and pass less data as a result.

**Review by yev-kanivets — APPROVED** [review]
Works like a charm 👍

**Pull request merged**